### ISSUE TITLE
your wallet uses less efficient method of querying electrs, consider contacting the developer of your wallet

### ISSUE THREAD

**Issue by yojoe** [trezor/trezor-suite#5267]
Using Trezor Suite [22.3.2](https://github.com/trezor/trezor-suite/releases/tag/v22.3.2) and electrs v0.9.6 as a custom Bitcoin backend I observe a lot of the following messages in the electrs log for `get_balance`, `listunspent` and `get_history` queries:

```
INFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.get_balance called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

```
NFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.listunspent called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

```
INFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.get_history called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

While Trezor Suite works with electrs, experience is very slow. I would expect some caching on recurring usage of the same Trezor device (wallet addresses) with electrs and I wonder if subscribing to the scripthashes first would improve the experience significantly? See a related comment: https://github.com/romanz/electrs/issues/608#issuecomment-960497420

**Comment by prusnak** [comment]
@marekrjpolak Have you tried subscribing to scripthash first before querying it? Or even better - subscribing to a set of scripthashes (let's say set size around 10-20) before querying them?

**Comment by marekrjpolak** [comment]
@prusnak Not yet, but it was discussed somewhere (I unfortunately cannot find it now) and it is planned. Initially I was a bit afraid that it would mean to subscribe even to somebody else's scripthashes (because of much richer data structure of Blockbook responses in comparison with Electrum) and I wasn't sure about the implied overhead, but after brief research it doesn't seem to be the case.

**Comment by Perlover** [comment]
Hello!

Today I downloaded the fresh version of Trezor Suite - 22.5.2
And I have own electrs (v0.9.7) server and I see right now in electrs logs:

.... your wallet uses less efficient method of querying electrs, consider contacting the developer of your wallet. Reason: blockchain.scripthash.get_history called for unsubscribed scripthash: XXXXX

Should this bug be fixed in 22.5.2 version? If yes why I can see same logs?

Thanks!

**Comment by hynek-jina** [comment]
This modification will be reflected in the version 22.6.1
